src/Tickets/Commerce/Gateways/Stripe/Settings.php

```diff
@@ -457,4 +457,33 @@ public function get_connection_settings_html() {
 		return $admin_views->template( 'settings/tickets-commerce/stripe/main', $context, false );
 	}
 
+	/**
+	 * Resets the values of payment methods and card options if they are no longer in use and avoid a settings
+	 * notice for empty values.
+	 *
+	 * @since TBD
+	 *
+	 * @param mixed  $value    Field value submitted.
+	 * @param string $field_id Field key in the settings array.
+	 * @param array  $field    Entire field array.
+	 *
+	 * @return mixed
+	 */
+	public function reset_hidden_field_values( $value, $field_id, $field ) {
+
+		if ( $value ) {
+			return $value;
+		}
+
+		if ( $field_id === static::$option_checkout_element_payment_methods ) {
+			return self::DEFAULT_PAYMENT_ELEMENT_METHODS;
```

Comment:
I think we could improve this by getting the option value here and returning it. That way if someone changes their mind and changes the checkout method they still have the same configuration.

Reply:
fixed